ecosystem/sep-0012.md

```diff
@@ -255,7 +271,16 @@ Content-Type: application/json
 {
    "account":"GBORFR3GDNVZ5PLUTBDQHKGWVD26CQUHORO2T3SDQ2JPLGLUJCCA5GK6",
    "memo":"21bf91a4-7db1-401d-8108-fab7660a45d6",
-   "memo_type":"text"
+   "memo_type":"text",
+   "photo_id_front": {
+      "data": "VGhpcyB0ZXh0IHJlcHJlc2VudHMgYmluYXJ5IGltYWdlIGRhdGE=",
+      "content_type": "image/png"
+   },
```

Comment:
This should be okay for smaller files, but this could be problematic for larger files because most JSON parsing libraries require the entire JSON object to be loaded into memory. It will be difficult for implementers to stream a file to some other destination, such as AWS S3.

What size files do you think we'll be handling?

This is more complicated, but it might be necessary to explore file upload as a separate API that returns a file identifier that is then provided here. This is a common pattern. e.g. https://stripe.com/docs/api/files https://developers.braintreepayments.com/reference/request/document-upload/create

Reply:
I agree with @leighmcculloch  that sending binary blobs in json is less than ideal. 

However, I'm concerned about the complexity of adding a file upload API to the spec.

What if we have the senders simply send a URL to the resource? this decouples the sending and receiving and allows both parties to utilize infrastructure that they (probably) already have. If we go down this route it's important to empathsize that these urls should be considered ephemeral so the receiver should fetch and save for their records.

Reply:
Using URLs would be harder to secure. These documents may be highly sensitive and submitting a URL that anyone could unauthenticated access isn't ideal. I think we need to define the method for the document to go from client to server without anyone else having access.